plugins/modules/inject_ks.py

```diff
@@ -121,188 +78,18 @@ def main():
             if not os.path.exists(module.params[key]):
                 module.fail_json("No such file found: %s" % module.params[key])
 
-    # define paths to things we need
-    isolinux_config = os.path.join(module.params["workdir"], "/isolinux/isolinux.cfg")
-    efi_grub_config = os.path.join(module.params["workdir"], "/EFI/BOOT/grub.cfg")
-    efi_dir = os.path.join(module.params["workdir"], "/EFI/BOOT")
-    efiboot_imagepath = os.path.join(module.params["workdir"], "/images/efiboot.img")
-
-    # get binary paths
-    ksvalidator = module.get_bin_path("ksvalidator")
-    isoinfo = module.get_bin_path("isoinfo")
-    xorriso = module.get_bin_path("xorriso")
-    mtype = module.get_bin_path("mtype")
-    mcopy = module.get_bin_path("mcopy")
-    mkefiboot = module.get_bin_path("mkefiboot")
-    genisoimage = module.get_bin_path("genisoimage")
-    isohybrid = module.get_bin_path("isohybrid")
-    implantisomd5 = module.get_bin_path("implantisomd5")
-    sed = module.get_bin_path("sed")
-
-    # validate the kickstart
-    ksvalidator_cmd = [
-        ksvalidator,
-        "-v",
-        module.params["version"],
-        module.params["kickstart"],
-    ]
-    ksvalidator_out = run_cmd(module, ksvalidator_cmd)
-
-    # get ISO volume id
-    isovolid_cmd = [isoinfo, "-d", "-i", module.params["src_iso"]]
-    isovolid_out = run_cmd(module, isovolid_cmd)
+    # Inject kickstart file to iso
     try:
-        isovolid = (
-            [line for line in isovolid_out.split("\n") if "Volume id:" in line][0]
-            .split(":")[-1]
-            .strip()
+        mkksiso_cmd = subprocess.run(
```

Comment:
Typically in Ansible modules, it's not recommended to call subprocess directly but instead use the provided wrapper from the `AnsibleModule` data type.

This is the pattern that's generally most common:

```python
    # get local for shelling out
    locale = get_best_parsable_locale(module)
    lang_env = dict(LANG=locale, LC_ALL=locale, LC_MESSAGES=locale)

    cmd_list = ["mkksiso", module.params["kickstart"], module.params["src_iso"], module.params["dest_iso"]]
    rc, out, err = module.run_command(cmd_list, environ_update=lang_env)
    if (rc != 0) or err:
        module.fail_json(
            "ERROR: Command '%s' failed with return code: %s and error message, '%s'"
            % (" ".join(cmd_list), rc, err)
        )
```